front/src/modules/timetableItem/components/Timetable/PacedTrain/PacedTrainItem.tsx

```diff
@@ -262,10 +285,8 @@ const PacedTrainItem = ({
           <div className="train-info">
             <span
               data-testid="paced-train-name"
-              className={cx(
-                'train-name',
-                `train-category-text-${TRAIN_MAIN_CATEGORY_CLASS[pacedTrain.category && isMainCategory(pacedTrain.category) ? pacedTrain.category.main_category : 'None']}`
-              )}
+              className={cx('train-name', getTrainCategoryClassName(pacedTrain.category, 'text'))}
```

Comment:
Even though we extracted the logic in a function, the class will be here if the category is a sub. We could probably do this (to test when your PR will be working). Same for the bg
```suggestion
              className={cx('train-name', {
                [getTrainCategoryClassName(pacedTrain.category, 'text')]:
                  category !== undefined && isMainCategory(category),
              })}
```

Reply:
Done